pkg/vf/vm.go

```diff
@@ -39,13 +39,39 @@ func NewVirtualMachine(vmConfig config.VirtualMachine) (*VirtualMachine, error)
 		vfConfig.SetPlatformVirtualMachineConfiguration(platformConfig)
 	} else {
 		PlatformType = "linux"
+		if vmConfig.NestedVirtualization {
+			err := setNestedVirtualization(vfConfig)
```

Comment:
Can we take a similar approach to the `macosBootloader` branch and introduce a `NewGenericPlatformConfiguration` function? At the moment, nested virtualization won’t be enabled with macOS guests.

Reply:
do you mean replace with `NewMacPlatformConfiguration` with `NewGenericPlatformConfiguration`?

Reply:
The `macOSBootloader` branch has to keep using `NewMacPlatformConfiguration`, but the linux codepath could make use of `NewGenericPlatformConfiguration` to have some kind of symmetry between the 2.